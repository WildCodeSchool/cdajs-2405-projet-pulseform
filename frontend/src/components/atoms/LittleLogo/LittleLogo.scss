@use "@assets/variables" as *;

.little-logo {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  @media (min-width: 768px) {
    display: flex;
    position: absolute;
    top: 40px;
    left: 40px;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }

  &--giant {
    width: 408px;
    height: 284px;
  }

  &--desktop {
    width: 124px;
    height: 78px;
  }

  &--mobile {
    width: 68px;
    height: 38px;
  }

  &__img {
    width: 100%;
    height: 100%;
  }

  &__title {
    color: $primary;
    margin: 0;

    &--giant {
      @include logo-big-label;
    }

    &--desktop {
      @include h1-phone;
    }

    &--mobile {
      @include logo-litle-label;
    }
  }
}
