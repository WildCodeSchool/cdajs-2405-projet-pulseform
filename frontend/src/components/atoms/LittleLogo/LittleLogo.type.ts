export interface LittleLogoProps {
  hasLabel?: boolean;
  size?: "mobile" | "giant" | "desktop";
  className?: string;
}
