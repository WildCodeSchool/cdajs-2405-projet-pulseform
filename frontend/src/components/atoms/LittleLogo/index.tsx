import classNames from "classnames";

import "./LittleLogo.scss";
import logoMobile from "@assets/icons/mobile-logo.svg";
import type { LittleLogoProps } from "./LittleLogo.type";

function LittleLogo({ hasLabel = false, size = "desktop" }: LittleLogoProps) {
  const littleLogoClassName = classNames("little-logo", {
    "little-logo--mobile": size === "mobile",
    "little-logo--giant": size === "giant",
    "little-logo--desktop": size === "desktop",
  });

  const littleLogoTitleClassName = classNames("little-logo__title", {
    "little-logo__title--mobile": size === "mobile",
    "little-logo__title--giant": size === "giant",
    "little-logo__title--desktop": size === "desktop",
  });

  return (
    <div className={littleLogoClassName}>
      <img
        className="little-logo__img"
        src={logoMobile}
        alt="Logo Pulse Form"
      />
      {hasLabel && <p className={littleLogoTitleClassName}>{"Pulse Form"}</p>}
    </div>
  );
}

export default LittleLogo;
