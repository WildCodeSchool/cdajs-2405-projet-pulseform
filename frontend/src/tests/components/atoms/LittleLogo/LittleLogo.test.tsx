import LittleLogo from "@components/atoms/LittleLogo";
import { render } from "@testing-library/react";

describe("LittleLogo Component", () => {
	it("renders the LittleLogo component without label", () => {
		const tree = render(<LittleLogo />);
		expect(tree).toMatchSnapshot();
	});

	it("renders the LittleLogo component with a label", () => {
		const tree = render(<LittleLogo hasLabel={true} />);
		expect(tree).toMatchSnapshot();
	});

	it("renders the LittleLogo giant component with a label", () => {
		const tree = render(<LittleLogo hasLabel={true} size="giant" />);
		expect(tree).toMatchSnapshot();
	});

	it("renders the LittleLogo mobile component with a label", () => {
		const tree = render(<LittleLogo hasLabel={true} size="mobile" />);
		expect(tree).toMatchSnapshot();
	});

	it("renders the LittleLogo giant component without label ", () => {
		const tree = render(<LittleLogo size="giant" />);
		expect(tree).toMatchSnapshot();
	});

	it("renders the LittleLogo mobile component without label", () => {
		const tree = render(<LittleLogo size="mobile" />);
		expect(tree).toMatchSnapshot();
	});
});
