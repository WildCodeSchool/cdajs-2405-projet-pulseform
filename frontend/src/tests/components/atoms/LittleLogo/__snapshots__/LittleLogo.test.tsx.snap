// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LittleLogo Component > renders the LittleLogo component with a label 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="little-logo little-logo--desktop"
      >
        <img
          alt="Logo Pulse Form"
          class="little-logo__img"
          src="/src/assets/icons/mobile-logo.svg"
        />
        <p
          class="little-logo__title little-logo__title--desktop"
        >
          Pulse Form
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="little-logo little-logo--desktop"
    >
      <img
        alt="Logo Pulse Form"
        class="little-logo__img"
        src="/src/assets/icons/mobile-logo.svg"
      />
      <p
        class="little-logo__title little-logo__title--desktop"
      >
        Pulse Form
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`LittleLogo Component > renders the LittleLogo component without label 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="little-logo little-logo--desktop"
      >
        <img
          alt="Logo Pulse Form"
          class="little-logo__img"
          src="/src/assets/icons/mobile-logo.svg"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="little-logo little-logo--desktop"
    >
      <img
        alt="Logo Pulse Form"
        class="little-logo__img"
        src="/src/assets/icons/mobile-logo.svg"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`LittleLogo Component > renders the LittleLogo giant component with a label 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="little-logo little-logo--giant"
      >
        <img
          alt="Logo Pulse Form"
          class="little-logo__img"
          src="/src/assets/icons/mobile-logo.svg"
        />
        <p
          class="little-logo__title little-logo__title--giant"
        >
          Pulse Form
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="little-logo little-logo--giant"
    >
      <img
        alt="Logo Pulse Form"
        class="little-logo__img"
        src="/src/assets/icons/mobile-logo.svg"
      />
      <p
        class="little-logo__title little-logo__title--giant"
      >
        Pulse Form
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`LittleLogo Component > renders the LittleLogo giant component without label  1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="little-logo little-logo--giant"
      >
        <img
          alt="Logo Pulse Form"
          class="little-logo__img"
          src="/src/assets/icons/mobile-logo.svg"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="little-logo little-logo--giant"
    >
      <img
        alt="Logo Pulse Form"
        class="little-logo__img"
        src="/src/assets/icons/mobile-logo.svg"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`LittleLogo Component > renders the LittleLogo mobile component with a label 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="little-logo little-logo--mobile"
      >
        <img
          alt="Logo Pulse Form"
          class="little-logo__img"
          src="/src/assets/icons/mobile-logo.svg"
        />
        <p
          class="little-logo__title little-logo__title--mobile"
        >
          Pulse Form
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="little-logo little-logo--mobile"
    >
      <img
        alt="Logo Pulse Form"
        class="little-logo__img"
        src="/src/assets/icons/mobile-logo.svg"
      />
      <p
        class="little-logo__title little-logo__title--mobile"
      >
        Pulse Form
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`LittleLogo Component > renders the LittleLogo mobile component without label 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="little-logo little-logo--mobile"
      >
        <img
          alt="Logo Pulse Form"
          class="little-logo__img"
          src="/src/assets/icons/mobile-logo.svg"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="little-logo little-logo--mobile"
    >
      <img
        alt="Logo Pulse Form"
        class="little-logo__img"
        src="/src/assets/icons/mobile-logo.svg"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
